src/components/vm/vmActions.tsx

```diff
@@ -63,140 +66,170 @@ import store from "../../store.js";
+                name: vm.name,
+                error: {
+                    text: cockpit.format(_("VM $0 failed to pause"), vm.name),
+                    detail: ex.message,
+                }
+            })
+        );
+    })
+);
+
+const onResume = (vm: VM) => (
+    domainResume({ id: vm.id, connectionName: vm.connectionName }).catch(ex => {
+        store.dispatch(
+            updateVm({
+                connectionName: vm.connectionName,
+                name: vm.name,
+                error: {
+                    text: cockpit.format(_("VM $0 failed to resume"), vm.name),
+                    detail: ex.message,
+                }
+            })
+        );
+    })
+);
+
+const onForceoff = (vm: VM) => (
+    domainForceOff({ id: vm.id, connectionName: vm.connectionName })
+            .then(() => !vm.persistent && cockpit.location.go(["vms"]))
+            .catch(ex => {
+                store.dispatch(
+                    updateVm({
+                        connectionName: vm.connectionName,
+                        name: vm.name,
+                        error: {
+                            text: cockpit.format(_("VM $0 failed to force shutdown"), vm.name),
+                            detail: ex.message,
+                        }
+                    })
+                );
+            })
+);
+
+const onSendNMI = (vm: VM) => (
+    domainSendNMI({ id: vm.id, connectionName: vm.connectionName }).catch(ex => {
+        store.dispatch(
+            updateVm({
+                connectionName: vm.connectionName,
+                name: vm.name,
+                error: {
+                    text: cockpit.format(_("VM $0 failed to send NMI"), vm.name),
+                    detail: ex.message,
+                }
+            })
+        );
+    })
+);
 
-const onSendNMI = (vm) => domainSendNMI({ name: vm.name, id: vm.id, connectionName: vm.connectionName }).catch(ex => {
-    store.dispatch(
-        updateVm({
-            connectionName: vm.connectionName,
-            name: vm.name,
-            error: {
-                text: cockpit.format(_("VM $0 failed to send NMI"), vm.name),
+const onAddTPM = (vm: VM, onAddErrorNotification: (n: Notification) => void) => (
+    domainAddTPM({ connectionName: vm.connectionName, vmName: vm.name })
+            .catch(ex => onAddErrorNotification({
+                text: cockpit.format(_("Failed to add TPM to VM $0"), vm.name),
                 detail: ex.message,
-            }
-        })
-    );
-});
-
-const onAddTPM = (vm, onAddErrorNotification) => domainAddTPM({ connectionName: vm.connectionName, vmName: vm.name })
-        .catch(ex => onAddErrorNotification({
-            text: cockpit.format(_("Failed to add TPM to VM $0"), vm.name),
-            detail: ex.message,
-            resourceId: vm.id,
-        }));
-
-const VmActions = ({ vm, vms, onAddErrorNotification, isDetailsPage = undefined }) => {
+                resourceId: vm.id,
```

Comment:
These 4 added lines are not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-2187-561408b0-20250717-145115-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)